lib/puppet/type/cisco_bgp.rb

```diff
@@ -216,25 +216,41 @@ def self.title_patterns
     end
   end # property confederation_id
 
-  newproperty(:confederation_peers) do
+  newproperty(:confederation_peers, array_matching: :all) do
     desc "AS confederation parameters. Valid values are String,
-          keyword 'default'."
+          keyword 'default' or Array."
 
     match_error = 'must be specified in ASPLAIN or ASDOT notation'
+
     validate do |peers|
-      list = peers.split(' ')
-      list.each do |value|
+      if peers.is_a? String
+        peers_array = peers.split(' ')
+      else
+        peers_array = peers
+      end
+      peers_array.each do |value|
         fail "Confederation peer value '#{value}' #{match_error}" unless
-          /^\d+$/.match(value) || /^\d+\.\d+$/.match(value) ||
-          peers == 'default' || peers == :default
+          /^(\d+|\d+\.\d+)$/.match(value) ||
+          peers_array == 'default' || peers_array == :default
       end
+      peers
```

Comment:
Do you need 'peers' at the end here? I don't think validate actually needs to return anything.